hip/CMakeLists.txt

```diff
@@ -144,8 +120,7 @@ list(APPEND CMAKE_PREFIX_PATH
 set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}" PARENT_SCOPE)
 set(CMAKE_PREFIX_PATH "${CMAKE_PREFIX_PATH}" PARENT_SCOPE)
 
-# setting the default flags like CMAKE_{LANG}_FLAGS_{TYPE}
-# the setting is copied from the default CMAKE_CXX_FLAGS_{TYPE}
+# NOTE: without this, HIP jacobi build takes a *very* long time
```

Comment:
Could you figure out why that is the case? Too low optimization level?

Reply:
Yes, AFAIK there is literally no flags passed to any compiler by default in HIP, thus by passing these flags we tell the HIP compilers to actually use optimization levels, which make things much better.
See: https://github.com/ROCm-Developer-Tools/HIP/blob/main/cmake/FindHIP.cmake#L9-L23

I wonder if we would get any benefit from passing in any device flag as well. I think for `nvcc` it doesn't matter, but I don't know about `clang`.